src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -1659,6 +1665,83 @@ def delete_deployment_script(cmd, resource_group_name, name):
     rcf.deployment_scripts.delete(resource_group_name, name)
 
 
+def get_template_spec(cmd, resource_group_name=None, name=None, version=None, template_spec=None):
+    if template_spec:
+        id_parts = parse_resource_id(template_spec)
+        resource_group_name = id_parts.get('resource_group')
+        name = id_parts.get('name')
+        version = id_parts.get('resource_name')
```

Comment:
Is this logic correct?

Reply:
The template spec resource ID could have `/versions/{version}` at the end, for example: `/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myRG/providers/Microsoft.Resources/templateSpecs/myTemplateSpec/versions/1.0`. If parsing such an ID with `parse_resource_id`, `resource_name` would be the version.